modules/l4subroute/handler.go

```diff
@@ -34,6 +35,9 @@ type Handler struct {
 	// The primary list of routes to compile and execute.
 	Routes layer4.RouteList `json:"routes,omitempty"`
 
+	// Maximum time connections have to complete the matching phase (the first terminal handler is matched). Default: 3s.
+	MatchingTimeout caddy.Duration `json:"matching_timeout,omitempty"`
```

Comment:
Would this be, like, 3s + 3s for the main route timeout + subroute? Or does the shorter of the two "win"?

Reply:
After the subroute handler is matched the deadline from the root handler is removed. Subroute will then start its own matching timeout. So yes it could be 3s + 3s. But if the subroute is not terminal we jump back to the main routing loop which resets the timeout to the old initial deadline. Which means it may immediately time out.

Reply:
oh ok, so if it spends its time matching in the subroute, it won't go back and add 3 more seconds in the main route. That's good then :+1: